src/Service/SparkPostService.php

```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
+            }
+        }
+
+        if ($message->getReplyTo()) {
+            $replyToList = $message->getReplyTo();
+            $replyToList->rewind();
+            $replyToAddress = $replyToList->current();
+
+            if ($replyToAddress instanceof AddressInterface) {
+                $metadata['reply_to'] = $replyToAddress->getName() ? $replyToAddress->toString() : $replyToAddress->getEmail();
+            }
+        }
+
+        return $metadata;
     }
 
     /**
      * Prepare body string from message
-     *
-     * @param  Message $message
-     *
-     * @return string
      */
     protected function prepareBody(Message $message): string
     {
         return $message->getBodyText();
     }
 
+    public function registerSendingDomain(string $domain): bool
+    {
+        $post = [
+            'domain' => urlencode($domain),
+            'dkim' => [
+                'public' => $this->dkimPublicKey,
+                'private' => $this->dkimPrivateKey,
+                'selector' => $this->dkimSelector,
+            ]
+        ];
+
+        $response = $this->prepareHttpClient('/sending-domains', $post)
+            ->send()
+        ;
+
+        // A 409-status means that the domains is already registered, which we consider a 'successful' result
+        $results = $this->parseResponse($response, [409]);
+
+        if($results && isset($results['results']) && isset($results['results']['message'])
+            && ($results['results']['message'] === 'Successfully Created domain.')) {
+            return true;
+        }
+
+        if ($response->getStatusCode() === 409) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function verifySendingDomain(string $domain): bool
+    {
+        $post = [
+            'dkim_verify' => true,
+        ];
+
+        $response = $this->prepareHttpClient(sprintf('/sending-domains/%s/verify', urlencode($domain)), $post)
+            ->send()
+        ;
+
+        $results = $this->parseResponse($response);
+
+        if($results && isset($results['results']) && isset($results['results']['dkim_status'])) {
+            return ($results['results']['dkim_status'] === 'valid');
+        } else {
+            return false;
+        }
+    }
+
+    public function removeSendingDomain(string $domain): void
+    {
+        $response = $this->prepareHttpClient(sprintf('/sending-domains/%s', urlencode($domain)))
+            ->setMethod(HttpRequest::METHOD_DELETE)
+            ->send()
+        ;
+
+        // When a 404-status is returned, the domains wasn't found, which is considered as a 'successful' response too
+        $this->parseResponse($response, [404]);
+    }
+
     /**
      * @param string $uri
      * @param array  $parameters
      * @return HttpClient
      */
-    private function prepareHttpClient(string $uri, array $parameters = []): HttpClient
+    protected function prepareHttpClient(string $uri, array $parameters = []): HttpClient
```

Comment:
Is there a use case for overriding this? I would think that _if_ you simply inject another client to the constructor of this service; to augment or extend the client itself.